public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -396,6 +397,15 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
     };
   }
 
+  metricFindQuery(query: string) {
+    if (!query) {
+      return Promise.resolve([]);
+    }
+    const interpolated = this.templateSrv.replace(query, {}, this.interpolateQueryExpr);
+    const metricFindQuery = new LokiMetricFindQuery(this, interpolated);
+    return metricFindQuery.process();
```

Comment:
Why is the `LokiMetricFindQuery` done as class? The statefulness does not seem to be useful  here. Seems like we could just have `processMetricFindQuery(this, interpolated)` and make it a simple function.